vllm/model_executor/layers/sampler.py

```diff
@@ -247,24 +255,30 @@ def _random_sample(
     selected_seq_groups: List[Tuple[List[int], SamplingParams]],
     is_prompts: List[bool],
     random_samples: torch.Tensor,
+    do_samples: List[bool],
 ) -> List[Tuple[List[int], List[int]]]:
     # Find the maximum best_of value of the prompt phase requests.
     random_samples = random_samples.cpu()
     sample_idx = 0
     results = []
-    for seq_group, is_prompt in zip(selected_seq_groups, is_prompts):
+    for seq_group, is_prompt, do_sample in zip(selected_seq_groups, is_prompts,
```

Comment:
most of this part is just indented (no logic change except that if do_sample is False, it doesn't do anything)